lib/Twig/Loader/Array.php

```diff
@@ -80,7 +80,7 @@ public function getCacheKey($name)
             throw new Twig_Error_Loader(sprintf('Template "%s" is not defined.', $name));
         }
 
-        return $this->templates[$name];
+        return $name.$this->templates[$name];
```

Comment:
you should add a delimiter between them, to avoid collision based on the beginning of the source and the end of the name